index.bs

```diff
@@ -308,15 +294,103 @@ The {{Blob()}} constructor can be invoked with the parameters below:
   <dd>which takes any number of the following types of elements, and in any order:
     * {{BufferSource}} elements.
     * {{Blob}} elements.
-    * {{DOMString}} elements.
+    * {{USVString}} elements.
 
   <dt id="dfn-BlobPropertyBagMembers">An *optional* {{BlobPropertyBag}}
-  <dd>which takes one member:
+  <dd>which takes two members:
     * <dfn id="dfn-BPtype" dict-member for="BlobPropertyBag">type</dfn>,
       the ASCII-encoded string in lower case representing the media type of the {{Blob}}.
       Normative conditions for this member are provided in the [[#constructorBlob]].
+    * <dfn dict-member for="BlobPropertyBag">endings</dfn>,
+      an enum which can take the values {{"transparent"}} or {{"native"}}.
+      By default this is set to {{"transparent"}}. If set to {{"native"}},
+      [=convert line endings to native|line endings will be converted to native=]
+      in any {{USVString}} elements in {{blobParts}}.
 </dl>
 
+<div algorithm="process-blob-parts">
+To <dfn lt="process blob parts|processing blob parts">process blob parts</dfn> given a sequence of {{BlobPart}}'s |parts|
+and {{BlobPropertyBag}} |options|,
+run the following steps:
+
+1. Let |bytes| be an empty sequence of bytes.
+
+1. For each |element| in |parts|:
+
+  1. If |element| is a {{USVString}}, run the following substeps:
+
+    1. Let |s| be |element|.
+
+    1. If the {{BlobPropertyBag/endings}} member of |options| is set to {{"native"}},
+      set |s| to the result of [=converting line endings to native=] of |element|.
+
+    1. Append the result of [=UTF-8 encoding=] |s| to |bytes|.
+
+      Note: The algorithm from WebIDL [[WebIDL]] replaces unmatched surrogates in an invalid utf-16 string
+      with U+FFFD replacement characters.
+      Scenarios exist when the {{Blob}} constructor may result in some data loss
+      due to lost or scrambled character sequences.
+
+  1. If |element| is a {{BufferSource}}, <a lt="get a copy of the buffer source">get
+    a copy of the bytes held by the buffer source</a>, and append those bytes to |bytes|.
+
+  1. If |element| is a {{Blob}},
+    append the bytes it represents to |bytes|.
+
+    Note: The {{Blob/type}} of the {{Blob}} array element is ignored and will not affect {{Blob/type}} of returned
+    {{Blob}} object.
+
+1. Return |bytes|.
+
+</div>
+
+<div algorithm="convert-line-endings-to-native">
+To <dfn lt="convert line endings to native|converting line endings to native">
+convert line endings to native</dfn> in a [=string=] |s|,
+run the following steps:
+
+1. Let |native line ending| be be the [=code point=] U+000A LF.
+
+1. If it matches the underlying platform's conventions,
+  set |native line ending| to the [=code point=] U+000D CR
+  followed by the [=code point=] U+000A.
```

Comment:
nit: Previous line has `U+000D CR` but this line has only `U+000A`

https://infra.spec.whatwg.org/#code-points says to use the name when it's not printable.

Super nitpicky: per http://www.unicode.org/charts/PDF/U0000.pdf it would be **U+000D CARRIAGE RETURN (CR)** and **U+000A LINE FEED (LF)** although Infra itself just uses LF

Reply:
I would prefer the shorter variant (we use those in Infra).

Reply:
Agreed that the shorter variant is clear enough, and seems to match what infra suggests. Added here, which I believe was the only place I was missing the name.